ebib-utils.el

```diff
@@ -2375,26 +2440,33 @@ year can be extracted from DATE, return nil."
   "Return the contents of FIELD from KEY in DB without TeX markup."
   (ebib-clean-TeX-markup (ebib-get-field-value field key db "" 'unbraced 'xref)))
 
+(defun ebib--convert-tex-italics (str)
+  "Return first sub-expression match in STR, in italics"
+  (propertize (match-string 1 str) 'face 'italic))
+
+(defun ebib--convert-tex-bold (str)
+  "Return first sub-expression match in STR, in bold"
+  (propertize (match-string 1 str) 'face 'bold))
+
+(defun ebib--convert-tex-small-caps (str)
+  "Return first sub-expression match in STR, capitalised"
+  (upcase (match-string 1 str)))
+
 (defun ebib-clean-TeX-markup (string)
-  "Return STRING without TeX markup."
-  (save-match-data
-    (let ((case-fold-search t))
-      ;; First replace TeX commands with their arguments and do some
-      ;; transformations, i.e., \textsc{cls} ==> CLS.  Note that optional
-      ;; arguments are also removed.
-      (while (string-match "\\\\\\([a-zA-Z*]*\\)\\(?:\\[.*?\\]\\)*{\\(.*?\\)}" string)
-        (let ((arg (copy-sequence (match-string 2 string))))
-          (pcase (match-string 1 string)
-            ((or "emph" "textit") (setq arg (propertize arg 'face '(italic))))
-            ("textbf" (setq arg (propertize arg 'face '(bold))))
-            ("textsc" (setq arg (upcase arg))))
-          (setq string (replace-match arg t t string)))))
-    ;; Replace as defined in `ebib-TeX-markup-replace-alist', and
-    ;; remove all {braces} (this takes care of nested braces too)
-    (replace-regexp-in-string
-     (rx-to-string (append '(or) (mapcar #'car ebib-TeX-markup-replace-alist)))
-     (lambda (match) (cdr (assoc match ebib-TeX-markup-replace-alist 'string=)))
-     string)))
+  "Return STRING without TeX markup.
+Any substring matching the car of a cell in
+`ebib-TeX-markup-replace-alist' is replaced with the
+correspnoding cdr (if the cdr is a string), or with the result of
+calling the cdr on the match (if it is a function). This is done
+with `replace-regexp-in-string', which see for details."
+  (let ((case-fold-search nil))
+    (save-match-data
+      (cl-loop for (pattern . replacement) in ebib-TeX-markup-replace-alist
+	       if (string-match pattern string)
```

Comment:
I don't think the `if` here is needed, since `replace-regexp-in-string` just returns the string unmodified if no matches are found.

Reply:
Fixed